### PR TITLE
Added dynamic fee in SetScriptTransaction

### PR DESCRIPTION
**Pull request by Anton-Rampage**

### CODE REVIEW

File: pkg/state/fee_validation.go [hunk 1]
```diff
@@ -36,6 +36,8 @@ var feeConstants = map[proto.TransactionType]uint64{
 	proto.InvokeExpressionTransaction: 5,
 }
 
+var SetScriptTransactionV6Fee uint64 = 1
```
Comment (alexeykiselev) [inline pkg/state/fee_validation.go:39]: May be change to `const`

File: pkg/state/fee_validation_test.go [hunk 2]
```diff
@@ -271,3 +272,46 @@ func TestSponsorshipFeeReduction(t *testing.T) {
...
+
+func randomScript(size uint64) (proto.Script, error) {
+	var s proto.Script = make([]byte, size)
+	_, err := rand.Read(s[:])
+	if err != nil {
+		return nil, err
+	}
+	return s, nil
+}
+
```
Comment (alexeykiselev) [inline pkg/state/fee_validation_test.go:284]: Please, add test on zero bytes script, it's a valid use case, setting empty script removes script from account. According to scala implementations setting zero size script should
cost 1 fee unit.

**Review by alexeykiselev — APPROVED** [review]

File: pkg/state/fee_validation.go [hunk 3]
```diff
@@ -140,6 +142,26 @@ func minFeeInUnits(params *feeValidationParams, tx proto.Transaction) (uint64, e
...
+		fee = SetScriptTransactionV6Fee
+		stx, ok := tx.(*proto.SetScriptWithProofs)
+		if !ok {
+			return 0, errors.New("failed to convert interface to SetScriptTransaction")
+		}
+
+		stxBytesForFee := len(stx.Script)
+		if stxBytesForFee < 0 {
+			panic(fmt.Sprintf("BUG, CREATE REPORT: setScriptTx bytes size (%d) must not be lower than zero", stxBytesForFee))
+		}
```
Comment (nickeskov) [inline pkg/state/fee_validation.go:162]: Actually, this check is redundant. Length can't physically be lower than zero.

**Review by nickeskov — CHANGES_REQUESTED** [review]

**Review by nickeskov — APPROVED** [review]
